core/src/cost_model.rs

```diff
@@ -36,6 +38,22 @@ pub struct TransactionCost {
     pub execution_cost: u64,
 }
 
+impl TransactionCost {
+    pub fn new_with_capacity(capacity: usize) -> Self {
+        Self {
+            writable_accounts: Vec::with_capacity(capacity),
+            account_access_cost: 0,
+            execution_cost: 0,
```

Comment:
you can replace these 2 lines with:
```
..Self::default()
```

Reply:
Makes sense ... an old habit to make things explicit ... am adapting into Rust concise manner. Same as above, will change in next refactoring.